src/snowflake/snowpark/mock/mock_plan.py

```diff
@@ -176,24 +177,12 @@ def calculate_expression(
     input_data: Union[TableEmulator, ColumnEmulator],
     analyzer,
 ) -> ColumnEmulator:
-    if isinstance(exp, UnresolvedAttribute):
-        return analyzer.analyze(exp)
+    if isinstance(exp, (UnresolvedAttribute, Attribute)):
+        return input_data[exp.name]
```

Comment:
exp of UnresolvedAttribute and Attribute won't need further analyze?

Reply:
My understanding is yes. "a" and "b" are UnresolvedAttribute in `df.select("a", "b")`.

Reply:
I see, let's keep this now